### PR TITLE
Assorted Dockerfile improvements

### PR DESCRIPTION
**Pull request by ncaio**
+ Organization
+ Declaration of variables ( _dockerfile, _dockertemp, _parameter1, _parameter2, _dockerdir)
+ Error handling

### CODE REVIEW

**Comment by Gsantomaggio** [comment]
Hi @ncaio  thank you for your PR.

I tried the your script on `Ubuntu 14.04.3 LTS` and I got this error:

```

./build-docker-image.sh 7
cp: cannot create regular file ‘docker-centos-7/Dockerfile’: No such file or directory
sed: can't read docker-centos-7/Dockerfile: No such file or directory
unable to prepare context: unable to evaluate symlinks in context path: lstat /home/gabriele/github/Pivotal/erlang-rpm/docker/docker-centos-7: no such file or directory
```

File: docker/build-docker-image.sh [hunk 1]
```diff
@@ -1,9 +1,58 @@
...
-rm -rf $DIR_DOCKER_FILE
-mkdir  $DIR_DOCKER_FILE
-cp Dockerfile.template  $DIR_DOCKER_FILE/Dockerfile
-sed -i 's/{centosfrom}/'$1'/g' $DIR_DOCKER_FILE/Dockerfile
-sudo docker build $2 -t="erlang-rpm-build-"$1 $DIR_DOCKER_FILE 
-
+#
+#
+#
+_dockerfile="Dockerfile"
```
Comment (Gsantomaggio) [inline docker/build-docker-image.sh]: can you please use standard naming convention ?
this should be `docker_file`

File: docker/build-docker-image.sh [hunk 2]
```diff
@@ -1,9 +1,58 @@
...
+_dockertemp="Dockerfile.template"
```
Comment (Gsantomaggio) [inline docker/build-docker-image.sh]: `temp` sounds like a temporary file. it should be `docker_template`

File: docker/build-docker-image.sh (hunk 1 above)
Comment (ncaio) [inline docker/build-docker-image.sh]: yes i can

File: docker/build-docker-image.sh (hunk 2 above)
Comment (ncaio) [inline docker/build-docker-image.sh]: thanks, good observation. in this case the copy process will be change to cp  "$_dockertemp" "$_dockerdir"/Dockerfile

**Comment by michaelklishin** [comment]
@ncaio you can push to the existing branch and GitHub will update the pull request instead of opening a new one. We'd be happy to accept your PR when all the points by @Gsantomaggio are addressed :)

**Comment by ncaio** [comment]
I closed because i submitted a new PR in/via branch. In the new PR i changed all @Gsantomaggio  points. by the way, @Gsantomaggio  thanks for the consideration. For the next's PR, we'll could apply a DISTRO condition in code.

Hi @ncaio thank you for your PR.
I tried your script on Ubuntu 14.04.3 LTS and I got this error:

This process is only for RPM like, basically RedHat and Centos. According erlang/rabbitmq Wiki 
 PS: Maybe will need a new Dockerfile.Template for Debian/Ubuntu.

let me know more about what you need. thank for your attention 

Ops: Do you prefer that i reopen this PR or not ? =]

**Comment by michaelklishin** [comment]
@ncaio I don't see any open PRs for this repository. Are you sure you have finished submitting one?

**Comment by michaelklishin** [comment]
I don't mind having a separate PR but if it's more convenient, simply push all your changes to the same branch and re-open this PR, GitHub should update it when you branch is advanced.

**Pull request closed without merging**